### PR TITLE
Fixed github #95: Allow using Pygments styles directly in .pdbrc.py

### PR DESCRIPTION
**Pull request by vaughnkoch**

### CODE REVIEW

**Comment by GHPS** [comment]
Great feature - works fine for me. Please merge the PR.

Before merging I suggest simplifying the usage from
`style = pygments.styles.get_style_by_name('monokai')`
to 
`style = 'monokai'`
since we are referring to a known set of style templates[1].

1: List of currently available style templates.
https://help.farbox.com/pygments.html

**Comment by GHPS** [comment]
I did some testing with the available styles. The results are a bit mixed: All themes with dark background work fine (native, friendly, monokai). Some styles have wrong colors (vim, bw) or show strange artifacts (tango).

It's a bit tedious to nail down the cause of this problem - in Kubuntu 16.04 it could be Linux, the terminal konsole or the pygments library.

@vaughnkoch: It would be nice if you could check styles other than monokai and see if the lighter themes work on your machine.

**Comment by blueyed** [comment]
@vaughnkoch 
Please address the comments above, and then rebase this on master, please.

I have not tested this myself yet, but thanks for the investigation/testing already, @GHPS !

File: pdb.py [hunk 1]
```diff
@@ -312,8 +312,11 @@ def _init_pygments(self):
                          if self.config.use_terminal256formatter
                             and '256color' in os.environ.get('TERM', '')
                          else TerminalFormatter)
-            self._fmt = Formatter(bg=self.config.bg,
-                                  colorscheme=self.config.colorscheme)
+            if self.config.use_terminal256formatter:
+                self._fmt = Formatter(style=self.config.style)
+            else:
+                self._fmt = Formatter(bg=self.config.bg,
+                                      colorscheme=self.config.colorscheme)
```
Comment (blueyed) [inline pdb.py:319]: Ah, so `style` is only for the `Terminal256Formatter` (had to look that up in the code though).
What about setting the formatter itself in the first place, which is already supported?

The current approach breaks if you would use the 256-color formatter, but `style` is not configured.
Maybe it should force the 256-color formatter if `style` is configured?

(just did a PR in this area myself - https://github.com/antocuni/pdb/pull/112)

**Comment by blueyed** [comment]
Ping?

**Comment by GHPS** [comment]
@blueyed - thanks for the time you spend on implementing this PR.

Styling is really a great idea - people loved it when this feature was
implemented in iPython [1].

But I am not sure what the current status of this PR is. Is the
feature ready to use?

1: https://github.com/ipython/ipython/issues/9541

**Comment by blueyed** [comment]
@GHPS 
I think you can achieve everything already via the dynamic config..  I've just pinged to see what the state / opinion(s) on this are.

**Comment by blueyed** [comment]
See https://github.com/antocuni/pdb/pull/222.

**Comment by GHPS** [comment]
@blueyed Thanks for implementing this great feature!

**Pull request closed without merging**